### DISCUSSION TITLE
Public or proxy endpoint Clarification

### DISCUSSION THREAD

**Discussion by BJDelaCruz97** [aws/graph-explorer#235] [Q&A]
Hi, 
Sorry but, can anyone clarify why do we need the server as public? as stated here,

> Public or proxy endpoint: Provide the publicly accessible endpoint URL for a graph database, e.g., Gremlin Server. If connecting to 
> Amazon Neptune, then provide a proxy endpoint URL that is accessible from outside the VPC, e.g., EC2.

Our Setup is AWS Neptune on Private VPC, cant be accessed outside.
For Graph Explorer, i installed in on EC2 within save vpc of neptune db.
Now i cant Synchronize the connection in Graph Explorer, but it is expected, just wanna know why we need the neptune db host publily accessible. just new with graph dbs sorry.

**Comment by triggan** [comment]
A Neptune cluster cannot be made publicly accessible.  It is only accessible within the VPC where it is hosted.  The purpose of Graph Explorer's proxy is to provide connectivity from the machine where you are accessing Graph Explorer into the VPC where Neptune is hosted.   When using the proxy, the data path is (local browser)->(GE proxy)->(Neptune cluster).  If you were using a VPN connection or Direct Connect connection to gain access to your VPC, only then could you bypass using the Graph Explorer proxy and connect to Neptune directly.  In the latter case, the connection path would be (local browser)->(Neptune cluster).
